src/Microsoft.DotNet.Wpf/src/PresentationCore/MS/internal/FontCache/FamilyCollection.cs

```diff
@@ -464,19 +465,16 @@ private CompositeFontFamily LookUpUserCompositeFamily(string familyName)
         /// <param name="fontFamily">The font family to look in.</param>
         /// <param name="faceName">The face to look for.</param>
         /// <returns>The font face if found and null if nothing was found.</returns>
-        private static Text.TextInterface.Font GetFontFromFamily(Text.TextInterface.FontFamily fontFamily, string faceName)
+        private static Text.TextInterface.Font GetFontFromFamily(Text.TextInterface.FontFamily fontFamily, ReadOnlySpan<char> faceName)
         {
-            faceName = faceName.ToUpper(CultureInfo.InvariantCulture);
-
             // The search that DWrite supports is a linear search.
             // Look at every font face.
             foreach (Text.TextInterface.Font font in fontFamily)
             {
                 // and at every locale name this font face has.
                 foreach (KeyValuePair<CultureInfo, string> name in font.FaceNames)
                 {
-                    string currentFontName = name.Value.ToUpper(CultureInfo.InvariantCulture);
-                    if (currentFontName == faceName)
+                    if (faceName.Equals(name.Value, StringComparison.OrdinalIgnoreCase))
```

Comment:
I am concerned about switching from invariant to ordinal. Using NLS, this no longer matches for example "ss" with "ß".

Not only that, I don't quite understand why the CultureInfo associated with that face name (i.e. name.Key) shouldn't be used for this comparison?

Applies to the dictionary too.

Reply:
I'm not sure it is that big of a concern, it was already using ordinal comparison but using InvariantCulture for the "ignore case" part. So the only thing that this PR could change is in the case where `faceName.Equals(name.Value, StringComparison.OrdinalIgnoreCase) != faceName.ToUpper(CultureInfo.InvariantCulture).Equals(name.Value.ToUpper(CultureInfo.InvariantCulture), StringComparison.Ordinal)`. Are you aware of any cases where this would apply ?

Reply:
just be aware of this: http://www.moserware.com/2008/02/does-your-code-pass-turkey-test.html 

(and I thought temporary fonts can have crazy (random) names).